File: app/src/main/java/vn/edu/usth/minigh/SplashActivity.java
package vn.edu.usth.minigh;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;

import java.util.Timer;
import java.util.TimerTask;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
    }

    @Override
    protected void onStart() {
        super.onStart();
        final SessionManagement sessionManagement = new SessionManagement(getApplicationContext());
        new Timer().schedule(new TimerTask() {
            @Override
            public void run() {
                int userID = sessionManagement.getSession();
                if(userID != -1) redirectActivity(ProfileActivity.class);
                else redirectActivity(AuthActivity.class);
            }
        }, 2000);
    }

    public void redirectActivity(Class aclass) {
        Intent intent = new Intent(this, aclass);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        startActivity(intent);
    }
}

File: app/src/main/java/vn/edu/usth/minigh/SplashActivity.kt
package vn.edu.usth.minigh

import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

import android.content.Intent
import android.os.Bundle

import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope

class SplashActivity : AppCompatActivity(R.layout.activity_splash) {
    override fun onStart() {
        super.onStart()

        lifecycleScope.launch {
            delay(2000)
            val manager = SessionManager(getApplicationContext())
            val intent = if (manager.getSession() == -1) {
                Intent(this@SplashActivity, AuthActivity::class.java)
            } else {
                Intent(this@SplashActivity, ProfileActivity::class.java)
            }
            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
            startActivity(intent)
        }
    }
}
